### ISSUE TITLE
Setting up a Go project with VSCode and devcontainer

### ISSUE THREAD

**Issue by avraam-inside** [golang/go#52134]
Hi!

I have a need to organize a workspace on one local machine with two operating systems, several databases, several programming languages, etc.

I am used to managing all this zoo with the help of containers.

Accordingly, I am trying to organize a Go project in VSCode fundamentally in dockers.

Unfortunately, I don't understand how I can organize the project so that VSCode doesn't throw errors, the Go code works out and it was clear to me as a person where the modules are, where the sources are, etc.

After reading such a guide https://go.dev/doc/tutorial/call-module-code and after poking, I came to an option that works in Go, but which VSCode doesn't like:

1. Go installed in /usr/local/go
1. PATH=${PATH}:/usr/local/go/bin
1. GOROOT=/usr/local/go
1. File structure
    ![image](https://user-images.githubusercontent.com/52919010/161546568-f8c5f48d-221a-4c57-b857-e012cef530c3.png)

Here is the worked code and errors from VSCode, which it is unclear what is being asked of me...
![image](https://user-images.githubusercontent.com/52919010/161547227-030ae8f8-90d6-426c-84af-6820285cfae2.png)

![image](https://user-images.githubusercontent.com/52919010/161547370-f38a2c12-0f24-45e5-8e13-d3dc10a34404.png)

Question: why does the Go code work out, and VSCode complains? What does VsCode want from me?

**Comment by seankhliao** [comment]
Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and proposals only.

For questions please refer to https://github.com/golang/go/wiki/Questions

**Comment by avraam-inside** [comment]
**In short**: the problem is solved.

I wouldn't say that this case is simple.

**Description**:

I'm new to the language, the first guides give out information about some GOPATH and GOROOT.

Further, upon request, the creation of the first project and the package project, some guides recommended writing code by defining GOPATH, somewhere GOPATH is missing in the documentation as such, somewhere it is mentioned crookedly.

And I needed to organize a workspace in a container with a clear understanding of what and where related to the language is located.

We've been racking our heads with colleagues for a decent amount of time.

**Decision:**

GOPATH, GOROOT and the project space are different entities that should be located in different locations.

As soon as I removed GOPATH from the root of the project, VSCode stopped complaining about the presence of go.mod.

**What could be done:**

@seankhliao The error "go.mod exists but should not be" unreadable for a beginner, at all. Indirectly, for me it turned out to be a loss of several days and nerves, for colleagues too.

Alternatively, modify the error description. And documentation https://go.dev/doc/tutorial/create-module by writing about what GOROOT is (including activation in the system via `PATH=${PATH}:/usr/local/go/bin`) and GOPATH, which needs to be kept separate from the project and GOROOT (which goes into confrontation with previous concepts, before the introduction of the default use of `go.mod`).
